### ISSUE TITLE
v5 alpha .nav-link transition and flashing borders

### ISSUE THREAD

**Issue by marcus-at-localhost** [twbs/bootstrap#31145]
In v5alpha I noticed a flashing of the tab borders on loading the page and I think it's related to the added transitions: https://github.com/twbs/bootstrap/pull/29988

I added a testcase, but also let me add a gif[1] and a video:
![bootstrap5](https://user-images.githubusercontent.com/1337223/85265301-5c896100-b472-11ea-810f-4528353da121.gif)

Full video: <https://imgur.com/a/LR0e1zT>

I reload the page, the tabs are being rendered with all blue borders and then the transparent border is being applied, with a transition (I throttled the connection/CPU in Dev Tools).

I can reproduce it by toggling the border and when I remove the transition from `.nav-link` there is no flashing anymore.

The gif and the video was made in a Chrome browser that's filled to the brim with extensions! I tested the same on a clean Chrome browser and the effect was much less visible and I couldn't capture it with the screenrecorder, but it's there!

- Chrome 83 Windows
- Testcase https://codepen.io/localhorst/pen/GRoWYRW

[1]: it looks like github remove the animation or just plays it once?!.

**Comment by seb-witt** [comment]
Same issue observed, additionally this seems to happen on input borders as well
Attached is behavour on the tabs as well, although styled, its still done with source sass from a few days ago

![3f915112-59d8-4e43-8459-19012a5d7c4c](https://user-images.githubusercontent.com/20238888/90040424-2de87380-dcc8-11ea-97a8-31b8270d00b4.gif)

Chrome: 84
And only plugin is adblock

For me it seems the faster you page rendering is, the more visible this becomes, which is quite a problem obtimizing for webvitals

**Comment by XhmikosR** [comment]
@twbs/css-review this might be something we have to tackle soon

**Comment by XhmikosR** [comment]
I updated to beta1 but I'm having trouble noticing anything in the CodPen: https://codepen.io/XhmikosR/pen/MWjomgq

**Comment by ffoodd** [comment]
I'm unable to reproduce this in Chormium 87, even with low-end mobile connection.

**Comment by marcus-at-localhost** [comment]
Hey, I just checked the codepen and the flashing is gone. Now in my original use case it's still there (latest v5beta).
What's happening there is a lot of vue.js/ajax/iframe loading, so I don't know if it's triggered by a re-render of the whole dom.

The closest I came to recreate the effect is here: https://codepen.io/localhorst/pen/XWjgeWO?editors=1010

I add the `nav-tabs` class after an initial delay (and even without delay - timeout 0) and that renders the tab borders first, until the transition to transparent borders is finished.

I don't know if this is too far fetched and an edge case.

**Comment by MartijnCuppens** [comment]
> I don't know if this is too far fetched and an edge case.

The flashing is expected behaviour then :p

https://github.com/twbs/bootstrap/issues/31145#issuecomment-672976262 also seems to be an issue with non-renderblocking CSS.

**Comment by marcus-at-localhost** [comment]
I was investigating this more, because it was bothering me. 
Here is how you can hopefully reproduce the bug:

TL;DR this only occurs if there is no `script` tag in the page.

1. Use Chrome Windows (I experienced it in Firefox Windows as well)
2. Fork this Codepen: https://codepen.io/localhorst/pen/poEdwvK
3. Change view to "Debug mode" so no codepen scripts are inserted
4. Open Devtools and check Disable Cache in Network tab
5. Load/Reload and the transition is flashing (not always but most of the time).

The reason why it was not working in codepen's Editor View in @XhmikosR example was
a) because Codepen injects extra `script` tags and 
b) @XhmikosR included the bootstrap JS

So it is only flashing if there are no `<script>` tag on the page and the css is external. Inlining bootstrap would fix  it.

This should show the flashing:

```html
<!DOCTYPE html>
<html lang="en" >
<head>

  <meta charset="UTF-8">
  <title>CodePen - Bootstrap 5 flashing Tabs</title>
  <meta name="viewport" content="width=device-width, initial-scale=1">
  
  <link rel='stylesheet' href='https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta1/dist/css/bootstrap.min.css'>
</head>
<body translate="no" >
  <nav>
  <div class="nav nav-tabs" id="nav-tab" role="tablist">
    <a href="#tab-start" data-bs-toggle="tab" role="tab" class="nav-link active">Start</a>
    <a href="#tab-console" data-bs-toggle="tab" role="tab" class="nav-link">php console</a>
    <a href="#tab-wiki" data-bs-toggle="tab" role="tab" class="nav-link">Wiki</a>
    <a href="#tab-wakatime" data-bs-toggle="tab" role="tab" class="nav-link">Wakatime</a>
    <a href="#tab-gistnote" data-bs-toggle="tab" role="tab" class="nav-link">Gistnote</a>
    <a href="#tab-startpage" data-bs-toggle="tab" role="tab" class="nav-link">OSCINT Startpage</a>
  </div>
</nav>
</body>
</html>
```

**The fix** is to put an "almost empty" `<script>` tag before `</head>` like this:

```html
<head>
  <link rel='stylesheet' href='https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta1/dist/css/bootstrap.min.css'>
  <script>//</script>
</head>
```

There are instances where a script tag in the body fixed it for people, there is a bug report for chrome related to input fields and transitions on load. In this particular case it only worked with empty script in the header.

This post https://stackoverflow.com/questions/14389566/stop-css-transition-from-firing-on-page-load helped me fixing it, even if it's all over the place.

I know this is nothing Bootstrap specific, but I thought it's maybe good to have it document here.
